src/controllers/interfaces.ts

```diff
@@ -0,0 +1,7 @@
+import { IAddress } from "../interface";
+
```

Comment:
Indeed, here we hold interfaces to be used within controllers (we don't have the "package protected" modifier in TypeScript, unfortunately).

We've somehow inherited this pattern from Go codebases, but seems applicable in TypeScript as well - works fine for languages with structural typing (as opposed to languages nominal-typing only - C# / Java).

Nice stuff to read & debate (not necessarily off-topic): Go interfaces vs. Java interfaces; nominal typing vs. structural typing.